locator/tablet_metadata_guard.hh

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "replica/database.hh"
+#include "locator/tablets.hh"
+#include "locator/abstract_replication_strategy.hh"
+
+namespace locator {
+
+/// A holder for table's effective_replication_map which
+/// automatically switches to the latest one as long as it
+/// does not affect the tablet associated with this guard.
+///
+/// This is useful for tracking long-running operations in
+/// the context of a single tablet which we don't want to
+/// block topology barriers for other tablets, only barriers for
+/// this tablet.
+class tablet_metadata_guard {
```

Comment:
@asias please be aware of this, it's needed to make repair not block tablet migrations.